### DISCUSSION TITLE
Allow a code owners' PRs to be approved by non-owners

### DISCUSSION THREAD

**Discussion by viettrungluu** [community/community#9636] [Pull Requests]
With GitHub's CODEOWNERS, every PR must be reviewed/approved by applicable owners, even when the author of the PR is themself an owner. Traditionally (e.g., with Chromium's and Google's OWNERS), an owner can get approval from anyone (and only needs owners approval for files they don't own).

The request is to allow the looser Chromium/Google system to be followed (at what granularity -- file/repository/branch -- is up for discussion).

## Rationale

This is extremely invaluable for onboarding new reviewers/owners. In the Chromium/Google/etc. system, new team members can review owners' code and be useful (there's a social contract on owners ensuring that their code is adequately reviewed; typically, easier/more trivial PRs are sent to new members for review first). In GitHub's system, new team members' reviews are fairly useless until they are made code owners. This leads to new members being made code owners too early without enough onboarding.

## Examples

Example 1: If I post a PR for files owned by team A and I'm a member of A:
* with GitHub, I need a member of A to approve
* with Chromium/Google, I still need approval from *someone*, but not necessarily a member of A (there's a social contract about ensuring that the review is adequate, but since I'm a member of A it should be in my interest to enforce that contract)

Example 2: If I post a PR for files owned by teams A and B and I am a member of A (but not B):
* with GitHub, I still need members of A and B to approve
* with Chromium/Google, I only need a member of B to approve (again, there's a social contract about ensuring that the files owned by A are appropriately reviewed)

**Comment by JonZeolla** [comment]
Related to this, if you have an item in CODEOWNERS which has only a single team assigned, and that team only has a single member at the moment, and that team member creates a PR, it is impossible to get that PR merged if you require CODEOWNER review without an admin override.  With the above proposed solution this issue would also be resolved.

**Reply by nkukushkin** [reply]
+1, this seems like an oversight

**Reply by Ro-M** [reply]
Yes, we are facing such a situation in our company as well. Ideally we want to avoid concrete usernames  in the CODEOWNER file and only add team names there. We often have the situation where a team only has one iOS developer so as a workaround we are adding another team as code owner which is as close to the original team's code as possible. But ideally this would not be needed.

**Reply by jakerockland** [reply]
+++ this also impacts us.

**Reply by vayan** [reply]
yeah it's pretty annoying to require an admin override :/

**Reply by jakerockland** [reply]
> yeah it's pretty annoying to require an admin override :/

+++, it encourages bad infosec policy practices IMO

**Reply by bdrazen** [reply]
How has this feature been out with this issue since 2017 and still not been addressed? 😕

**Reply by sam-utila** [reply]
GitHub is not built for enterprise

**Reply by sanchay-hai** [reply]
GitHub this practically makes the CODEOWNERs useless. Please fix this

**Comment by oncilla** [comment]
This feature would be very useful for our organization :+1:

**Comment by zeroliu** [comment]
This issue happens for us daily. We are a small engineering team, but it requires ownership to prevent unauthorized changes every day. Code ownership is essential and we can't use CODEOWNERs because it lacks this fundamental functionality. I can't believe this hasn't been implemented since 2022. This feature is a must-have; without it, code ownership is completely useless.
